### PR TITLE
catlog2lucene waits for one or more registration cursors

### PR DESCRIPTION
**Pull request by cristinamanum**
In order to support multiple regions registrations and one search service force catalog2Lucene to wait for all registrations.  The cursor used to advance catalog2lucene will have the value the min of all passed cursors.

### CODE REVIEW

File: src/Ng/Jobs/Catalog2LuceneJob.cs [hunk 1]
```diff
@@ -107,6 +107,13 @@ protected override void Init(IDictionary<string, string> arguments, Cancellation
             }
         }
 
+        ReadCursor GetTheLeastAdvancedRegistrationCursor(string registrationArg, CancellationToken cancellationToken)
```
Comment (joelverhagen) [inline src/Ng/Jobs/Catalog2LuceneJob.cs]: `private`

File: src/Ng/Jobs/Catalog2LuceneJob.cs [hunk 2]
```diff
@@ -107,6 +107,13 @@ protected override void Init(IDictionary<string, string> arguments, Cancellation
...
+        {
+            string[] registrations = registrationArg.Split(';');
```
Comment (scottbommarito) [inline src/Ng/Jobs/Catalog2LuceneJob.cs]: Any reason we can't just query index.json and get all of the registrations from there?

**Comment by skofman1** [comment]
:shipit:

File: src/Ng/Jobs/Catalog2LuceneJob.cs (hunk 1 above)
Comment (cristinamanum) [inline src/Ng/Jobs/Catalog2LuceneJob.cs]: Updated

File: src/Ng/Jobs/Catalog2LuceneJob.cs (hunk 2 above)
Comment (cristinamanum) [inline src/Ng/Jobs/Catalog2LuceneJob.cs]: Updated

Comment (cristinamanum) [inline src/Ng/Jobs/Catalog2LuceneJob.cs]: There will be one registration in index.json ; its location will be masked by a dns. The job will need to read the cursor blob location.

Comment (scottbommarito) [inline src/Ng/Jobs/Catalog2LuceneJob.cs]: After offline discussion, I'll file an issue regarding how we can harden our services to change with index.json changes.

**Pull request merged**